pilot/pkg/xds/ads.go

```diff
@@ -185,14 +173,14 @@ func (s *DiscoveryServer) receive(con *Connection, reqChannel chan *discovery.Di
 // handles 'push' requests and close - the code will eventually call the 'push' code, and it needs more mutex
 // protection. Original code avoided the mutexes by doing both 'push' and 'process requests' in same thread.
 func (s *DiscoveryServer) processRequest(req *discovery.DiscoveryRequest, con *Connection) error {
-
-	if !s.preProcessRequest(con.proxy, req) {
+	if req.TypeUrl == v3.HealthInfoType {
```

Comment:
I like the old code better, this leaks low level details into processRequest

Reply:
I would argue that TypeUrl should be a high level stuff, it's more like a switch case.

And also I donot like the return value `should continue`, lol